src/App.tsx

```diff
@@ -15,36 +16,113 @@
   'Garlic',
 ];
 
+enum SortType {
+  ABC = 'Sort alphabetically',
+  LENGTH = 'Sort by length',
+  RESET = 'Reset',
+}
+
+type DataFromServer = string;
+
+const handleSorting = function (
+  data: DataFromServer[],
+  sortType: SortType | null,
+  isReversed: boolean,
+) {
+  const sortedData = [...data];
+
+  if (sortType) {
+    sortedData.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.ABC:
+          return good1.localeCompare(good2);
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+        case SortType.RESET:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    sortedData.reverse();
+  }
+
+  return sortedData;
+};
+
 export const App: React.FC = () => {
+  const [selectedSortType, setSelectedSortType] = useState<SortType | null>(
+    null,
+  );
+  const [isReversed, setReversed] = useState<boolean>(false);
+
+  const visibleData = handleSorting(
+    goodsFromServer,
+    selectedSortType,
+    isReversed,
+  );
+
+  function handleReset() {
```

Comment:
The function's name should start with a verb to clearly indicate its purpose. Consider renaming `handleReset` to `resetSelection` or `clearSortAndReverse`.